extensions/preview/ui/component-preview/preview.tsx

```diff
@@ -32,15 +35,60 @@ export type ComponentPreviewProps = {
  * renders a preview of a component.
  */
 export function ComponentPreview({ component, style, previewName, queryParams }: ComponentPreviewProps) {
-  const serverUrl = `/api/${component.id.fullName}/@/preview`;
-
-  const url = `${(component.server && component.server.url) || serverUrl}/#${component.id.fullName}${
-    `?preview=${previewName}&${queryParams && queryParams}` || ''
-  }`;
+  const serverPath = toPreviewServer(component);
+  const hash = toPreviewHash(component, previewName, queryParams);
 
-  return <iframe style={style} src={url} />;
+  return <iframe style={style} src={`${serverPath}/#${hash}`} />;
 }
 
 ComponentPreview.defaultProps = {
   hotReload: true,
 };
+
+/**
+ * create a string segment
+ * append a prefix and a suffix to a string, only if defined
+ */
+function optional(prefix = '', str?: string, suffix = '') {
+  if (!str) return '';
+
+  return `${prefix}${str}${suffix}`;
+}
+
+/**
+ * generates preview server path from component data
+ */
+export function toPreviewServer(component: ComponentModel) {
+  // TODO  - check if this is needed
+  const defaultServerUrl = `/api/${component.id.fullName}/@/preview`;
```

Comment:
@ranm8 - I just want to verify this is still relevant. If so, it will create a different iframe for each component (!). So we might need to disable the cached iframes feature.